file-formats/ints/type/zif_aff_ints_v1.intf.abap

```diff
@@ -0,0 +1,278 @@
+
+  TYPES:
+    "! <p class="shorttext">Scenario Object</p>
+    "! Object generated based on a scenario
+    BEGIN OF ty_scenario_ddl_object,
+      "! <p class="shorttext">Object Name</p>
+      "! Object name
+      object_name         TYPE c LENGTH 40,
+      "! <p class="shorttext">Object Type</p>
+      "! Type of the object
+      object_type         TYPE c LENGTH 4,
+      "! <p class="shorttext">Object Status</p>
+      "! Status of the object
+      object_status       TYPE ty_object_status,
+      "! <p class="shorttext">Logical Object Type</p>
+      "! Logical object type of a BOM object
+      logical_object_type TYPE ty_logical_object_type,
+    END OF ty_scenario_ddl_object,
+
+    "! <p class="shorttext">Intelligent Scenario</p>
+    "! Intelligent scenario
+    BEGIN OF ty_intelligent_scenario,
+      "! <p class="shorttext">Intelligent Scenario Type</p>
+      "! Intelligent scenario type
+      "! $required
+      scenario_type              TYPE c LENGTH 256,
+      "! <p class="shorttext">Extensible</p>
+      "! If true, the scenario is extensible
+      is_extensible              TYPE abap_bool,
+      "! <p class="shorttext">Multiple Active Models</p>
+      "! If true, the scenario can have multiple active models
+      is_multi_active_model      TYPE abap_bool,
+      "! <p class="shorttext">Data Management</p>
+      "! If true, the scenario uses Data Management
+      is_data_management_enabled TYPE abap_bool,
+      "! <p class="shorttext">Automate Turnkey Switch On</p>
+      "! If true, scenario will be turnkey enabled to automate the operations
+      is_turnkey_auto_switch_on  TYPE abap_bool,
+      "! <p class="shorttext">Turnkey Implementation Type</p>
+      "! Type of turnkey implementation
+      turnkey_type               TYPE c LENGTH 10,
+      "! <p class="shorttext">Scenario Technology</p>
+      "! Scenario technology
+      scenario_technology        TYPE c LENGTH 20,
+      "! <p class="shorttext">Prediction Class</p>
+      "! Prediction class
+      prediction_class           TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Prerequisite Check Class</p>
+      "! Prerequisite check class
+      prerequisite_check_class   TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Scenario Technical ID</p>
+      "! Scenario technical ID
+      scenario_technical_id      TYPE c LENGTH 64,
+      "! <p class="shorttext">State</p>
+      "! State of an intelligent scenario
+      state                      TYPE c LENGTH 10,
+      "! <p class="shorttext">ISLM Framework Version</p>
+      "! ISLM framework version
+      islm_version               TYPE p LENGTH 4 DECIMALS 2,
+      "! <p class="shorttext">Turnkey Class</p>
+      "! Turnkey class
+      turnkey_class              TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Usage Type</p>
+      "! Usage type of an intelligent scenario
+      usage_type                 TYPE c LENGTH 10,
```

Comment:
I saw you made a value help for this field. Can we also have an enum here like we did for object_statusand logical_object_type?

Reply:
We are calling backend class method to get the list of possible values. So, enum is not possible.